src/processes/routing/types.ts

```diff
@@ -15,6 +16,32 @@ export type PaymentsNavigationProps = BottomTabScreenProps<
   'PaymentsNavigation'
 >
 
+export type AuthStackParams = {
+  PhoneAuth: undefined
+  AuthStatus: {
+    isSucceeded: boolean
+  },
+  Otp: undefined,
+  PasswordAuth: undefined
+}
+
+export type PhoneAuthScreenProps = NativeStackScreenProps<
+  AuthStackParams,
+  "PhoneAuth"
+>
+export type AuthStatusScreenProps = NativeStackScreenProps<
+  AuthStackParams,
+  "AuthStatus"
+>
+export type OtpScreenProps = NativeStackScreenProps<
+  AuthStackParams,
+  "Otp"
+>
+export type PasswordAuthScreenProps = NativeStackScreenProps<
+  AuthStackParams, 
+  "PasswordAuth"
+>
+
```

Comment:
Можно убрать дублирование

```suggestion
export type AuthStackParams = {
  PhoneAuth: undefined
  AuthStatus: {
    isSucceeded: boolean
  },
  Otp: undefined,
  PasswordAuth: undefined
}

type TAuthStackProps<T extends keyof AuthStackParams> = NativeStackScreenProps<AuthStackParams, T>;

export type PhoneAuthScreenProps = TAuthStackProps<"PhoneAuth">

export type AuthStatusScreenProps = TAuthStackProps<"AuthStatus">

export type OtpScreenProps = TAuthStackProps<"Otp">

export type PasswordAuthScreenProps = TAuthStackProps<"PasswordAuth">

```